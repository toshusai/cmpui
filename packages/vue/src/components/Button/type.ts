export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
