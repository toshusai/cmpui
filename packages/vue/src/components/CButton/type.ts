export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
