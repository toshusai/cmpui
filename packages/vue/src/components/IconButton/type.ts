export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
