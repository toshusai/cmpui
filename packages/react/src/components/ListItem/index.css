.cmpui_list-item__root {
  color: var(--cmpui-text-color);

  align-items: center;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  font-size: 14px;
  min-height: 32px;
  padding: 0 8px;
  width: 100%;
  cursor: pointer;
}

.cmpui_list-item__root:hover {
  background-color: var(--cmpui-bg-hover-color);
}

.cmpui_list-item__root:active {
  background-color: var(--cmpui-bg-active-color);
}

.cmpui_list-item__root[data-highlighted],
.cmpui_list-item__root[data-highlighted]:active,
.cmpui_list-item__root[data-highlighted]:active,
.cmpui_list-item__root[data-highlighted]:hover,
.cmpui_list-item__root:focus-visible {
  background-color: var(--cmpui-bg-focus-color);
  color: white;
  outline: none;
}

.cmpui_list-item__root[aria-selected="true"] {
  background-color: var(--cmpui-bg-focus-color);

  color: white;
}

.cmpui_list-item__root[aria-disabled="true"],
.cmpui_list-item__root[aria-disabled="true"]:hover,
.cmpui_list-item__root[aria-disabled="true"]:active {
  background-color: transparent;
  cursor: default;
  opacity: 0.7;
}

.cmpui_list-item__root[data-size="S"] {
  font-size: 12px;
  height: 24px;
  min-height: 24px;
}

.cmpui_list-item__root[data-rounded="true"] {
  border-radius: 4px;
}

.cmpui_list-list__root {
  margin: 0;
  padding: 0;
}
