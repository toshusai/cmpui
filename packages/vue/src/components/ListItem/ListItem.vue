<script setup lang="ts">
withDefaults(
  defineProps<{
    is?: string;
    rounded?: boolean;
    selected?: boolean;
    size?: "S" | "M";
  }>(),
  {
    is: "li",
    size: "M",
  },
);
</script>

<template>
  <component
    :is="is"
    class="cmpui_list-item__root"
    :data-size="size"
    :daat-rounded="rounded"
    :aria-selected="selected"
    v-bind="$attrs"
  >
    <slot></slot>
  </component>
</template>
