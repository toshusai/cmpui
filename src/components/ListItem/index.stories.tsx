import type { Meta, StoryObj } from "@storybook/react";
import { List, ListItem } from ".";
import { FloatBox } from "../FloatBox";

const meta: Meta<typeof ListItem> = {
  title: "View/ListItem",
  component: ListItem,
};

export default meta;
type Story = StoryObj<typeof ListItem>;

export const Basic: Story = {
  render: function RenderBasic() {
    return (
      <List as="ol">
        <ListItem>Item1</ListItem>
        <ListItem>Item2</ListItem>
        <ListItem>Item3</ListItem>
      </List>
    );
  },
};

export const Selected: Story = {
  render: function RenderBasic() {
    return (
      <List as="ol">
        <ListItem selected>Item1</ListItem>
        <ListItem>Item2</ListItem>
        <ListItem>Item3</ListItem>
      </List>
    );
  },
};

export const Small: Story = {
  render: function RenderBasic() {
    return (
      <List as="ol">
        <ListItem size="S">Item1</ListItem>
        <ListItem size="S">Item2</ListItem>
        <ListItem size="S">Item3</ListItem>
      </List>
    );
  },
};

function CommandItem(props: {
  children?: React.ReactNode;
  shortcut?: string;
  selected?: boolean;
}) {
  return (
    <ListItem
      style={{
        width: 128,
      }}
      rounded
      selected={props.selected}
      size="S"
    >
      {props.children}
      {props.shortcut && (
        <span
          style={{
            marginLeft: "auto",
            fontSize: 10,
            color: props.selected ? "white" : "var(--cmpui-text2-color)",
          }}
        >
          {props.shortcut}
        </span>
      )}
    </ListItem>
  );
}

export const Commands: Story = {
  render: function RenderBasic() {
    return (
      <FloatBox>
        <List as="ol">
          <CommandItem shortcut="⌘ Z">Undo</CommandItem>
          <CommandItem shortcut="⇧ ⌘ Z">Redo</CommandItem>
          <CommandItem shortcut="⌘ X">Cut</CommandItem>
          <CommandItem selected shortcut="⌘ C">
            Copy
          </CommandItem>
          <CommandItem shortcut="⌘ V">Paste</CommandItem>
        </List>
      </FloatBox>
    );
  },
};
