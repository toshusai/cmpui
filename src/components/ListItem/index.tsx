import React, { ForwardedRef, forwardRef } from "react";
import { CustomJSXElement } from "../../types/CustomJSXElement";
import { classNames } from "../../utils/classNames";

import "./index.css";

export type ListItemProps<T extends CustomJSXElement = "li"> = {
  as?: T;
  children: React.ReactNode;
  className?: string;
  rounded?: boolean;
  selected?: boolean;
  size?: "S" | "M";
} & React.ComponentProps<T>;

export const ListItem = forwardRef(function ListItem<
  T extends CustomJSXElement,
>(
  {
    as: Component = "li",
    children,
    className,
    rounded,
    size = "M",
    ...props
  }: ListItemProps<T>,
  ref: ForwardedRef<HTMLElement>,
) {
  return (
    <Component
      className={classNames("cmpui_list-item__root", className)}
      ref={ref}
      aria-selected={props.selected}
      data-size={size}
      data-rounded={rounded}
      {...props}
    >
      {children}
    </Component>
  );
}) as <T extends CustomJSXElement = "li">(p: ListItemProps<T>) => JSX.Element;

export const List = forwardRef<
  HTMLUListElement,
  React.ComponentProps<"ul"> & {
    as?: React.ElementType;
  }
>(({ className, children, as: Component = "ul", ...props }, ref) => {
  return (
    <Component
      className={classNames("cmpui_list-list__root", className)}
      ref={ref}
      {...props}
    >
      {children}
    </Component>
  );
});
