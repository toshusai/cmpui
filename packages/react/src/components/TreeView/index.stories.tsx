import type { Meta, StoryObj } from "@storybook/react";

import { ListItem } from "../ListItem";

import { TreeView } from ".";

const meta: Meta<typeof TreeView> = {
  title: "View/TreeView",
  component: TreeView,
};

export default meta;

type Story = StoryObj<typeof TreeView>;

type Item = {
  id: string;
  children?: Item[];
  text: string;
  isExpanded?: boolean;
};

export const Basic: Story = {
  render: function Render() {
    const items: Item[] = [
      {
        id: "1",
        text: "One",
      },
      {
        id: "2",
        text: "Two",
        children: [
          {
            id: "3",
            text: "Three",
            children: [
              {
                id: "5",
                text: "Five",
              },
            ],
          },
          {
            id: "6",
            text: "Six",
          },
        ],
      },
      {
        id: "4",
        text: "Four",
      },
    ];

    return (
      <div
        style={{
          height: 256,
          width: 256,
          overflow: "auto",
        }}
      >
        <TreeView<Item>
          items={items}
          render={(props) => (
            <ListItem style={{ paddingLeft: `${props.depth * 16 + 8}px` }}>
              {props.item.text}
            </ListItem>
          )}
        />
      </div>
    );
  },
};
