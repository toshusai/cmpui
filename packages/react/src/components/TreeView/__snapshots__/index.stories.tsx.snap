// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View/TreeView Basic smoke-test 1`] = `
<div data-state="closed"
     style="height: 256px; width: 256px; overflow: auto;"
>
  <div class="cmpui_tree-view__root">
    <ol class="cmpui_tree-view__ol">
      <li class="cmpui_tree-view__item">
        <div class="cmpui_list-item__root"
             aria-selected="false"
             data-size="M"
             data-state="closed"
             style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 8px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
        >
          <div style="display: flex; gap: 4px; cursor: pointer;">
            <span class="material-symbols-outlined"
                  style="font-size: 16px;"
            >
              expand_more
            </span>
            <span class="material-symbols-outlined"
                  style="font-size: 16px;"
            >
              folder_open
            </span>
          </div>
          root
        </div>
      </li>
      <li class="cmpui_tree-view__item-dir">
        <ol class="cmpui_tree-view__item-dir-ol">
          <li class="cmpui_tree-view__item">
            <div class="cmpui_list-item__root"
                 aria-selected="false"
                 data-size="M"
                 data-state="closed"
                 style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 16px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
            >
              <div style="display: flex; gap: 4px; cursor: pointer;">
                <span class="material-symbols-outlined"
                      style="font-size: 16px;"
                >
                  expand_more
                </span>
                <span class="material-symbols-outlined"
                      style="font-size: 16px;"
                >
                  folder_open
                </span>
              </div>
              src
            </div>
          </li>
          <li class="cmpui_tree-view__item-dir">
            <ol class="cmpui_tree-view__item-dir-ol">
              <li class="cmpui_tree-view__item">
                <div class="cmpui_list-item__root"
                     aria-selected="false"
                     data-size="M"
                     data-state="closed"
                     style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 24px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
                >
                  <span class="material-symbols-outlined"
                        style="font-size: 16px; margin-left: 12px;"
                  >
                    image
                  </span>
                  image.png
                </div>
              </li>
              <li class="cmpui_tree-view__item">
                <div class="cmpui_list-item__root"
                     aria-selected="false"
                     data-size="M"
                     data-state="closed"
                     style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 24px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
                >
                  <span class="material-symbols-outlined"
                        style="font-size: 16px; margin-left: 12px;"
                  >
                    music_note
                  </span>
                  music.mp3
                </div>
              </li>
            </ol>
          </li>
          <li class="cmpui_tree-view__item">
            <div class="cmpui_list-item__root"
                 aria-selected="false"
                 data-size="M"
                 data-state="closed"
                 style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 16px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
            >
              <div style="display: flex; gap: 4px; cursor: pointer;">
                <span class="material-symbols-outlined"
                      style="font-size: 16px;"
                >
                  expand_more
                </span>
                <span class="material-symbols-outlined"
                      style="font-size: 16px;"
                >
                  folder_open
                </span>
              </div>
              lib
            </div>
          </li>
          <li class="cmpui_tree-view__item-dir">
            <ol class="cmpui_tree-view__item-dir-ol">
              <li class="cmpui_tree-view__item">
                <div class="cmpui_list-item__root"
                     aria-selected="false"
                     data-size="M"
                     data-state="closed"
                     style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 24px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
                >
                  <span class="material-symbols-outlined"
                        style="font-size: 16px; margin-left: 12px;"
                  >
                    videocam
                  </span>
                  video.mp4
                </div>
              </li>
              <li class="cmpui_tree-view__item">
                <div class="cmpui_list-item__root"
                     aria-selected="false"
                     data-size="M"
                     data-state="closed"
                     style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 24px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
                >
                  <span class="material-symbols-outlined"
                        style="font-size: 16px; margin-left: 12px;"
                  >
                    code
                  </span>
                  code.tsx
                </div>
              </li>
            </ol>
          </li>
          <li class="cmpui_tree-view__item">
            <div class="cmpui_list-item__root"
                 aria-selected="false"
                 data-size="M"
                 data-state="closed"
                 style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 16px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
            >
              <div style="display: flex; gap: 4px; cursor: pointer;">
                <span class="material-symbols-outlined"
                      style="font-size: 16px;"
                >
                  expand_more
                </span>
                <span class="material-symbols-outlined"
                      style="font-size: 16px;"
                >
                  folder_open
                </span>
              </div>
              child3
            </div>
          </li>
          <li class="cmpui_tree-view__item-dir">
            <ol class="cmpui_tree-view__item-dir-ol">
              <li class="cmpui_tree-view__item">
                <div class="cmpui_list-item__root"
                     aria-selected="false"
                     data-size="M"
                     data-state="closed"
                     style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 24px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
                >
                  <span class="material-symbols-outlined"
                        style="font-size: 16px; margin-left: 12px;"
                  >
                    note
                  </span>
                  child3-1
                </div>
              </li>
              <li class="cmpui_tree-view__item">
                <div class="cmpui_list-item__root"
                     aria-selected="false"
                     data-size="M"
                     data-state="closed"
                     style="display: flex; align-items: center; width: 100%; height: 100%; padding-left: 24px; box-sizing: border-box; padding-right: 8px; gap: 4px;"
                >
                  <span class="material-symbols-outlined"
                        style="font-size: 16px; margin-left: 12px;"
                  >
                    note
                  </span>
                  child3-2
                </div>
              </li>
            </ol>
          </li>
        </ol>
      </li>
    </ol>
  </div>
</div>
`;

exports[`View/TreeView Scroll smoke-test 1`] = `
<div style="width: 256px; padding-bottom: 32px; overflow: auto;">
  <div class="cmpui_tree-view__root">
    <ol class="cmpui_tree-view__ol">
      <li class="cmpui_tree-view__item">
        <div style="background-color: rgba(0, 0, 0, 0.05); padding-left: 0px; white-space: nowrap;">
          root
        </div>
      </li>
      <li class="cmpui_tree-view__item-dir">
        <ol class="cmpui_tree-view__item-dir-ol">
          <li class="cmpui_tree-view__item">
            <div style="background-color: rgba(0, 0, 0, 0.1); padding-left: 16px; white-space: nowrap;">
              child-1
            </div>
          </li>
          <li class="cmpui_tree-view__item-dir">
            <ol class="cmpui_tree-view__item-dir-ol">
              <li class="cmpui_tree-view__item">
                <div style="background-color: rgba(0, 0, 0, 0.15); padding-left: 32px; white-space: nowrap;">
                  child-1-1
                </div>
              </li>
              <li class="cmpui_tree-view__item-dir">
                <ol class="cmpui_tree-view__item-dir-ol">
                  <li class="cmpui_tree-view__item">
                    <div style="background-color: rgba(0, 0, 0, 0.2); padding-left: 48px; white-space: nowrap;">
                      child-1-1-1
                    </div>
                  </li>
                  <li class="cmpui_tree-view__item-dir">
                    <ol class="cmpui_tree-view__item-dir-ol">
                      <li class="cmpui_tree-view__item">
                        <div style="background-color: rgba(0, 0, 0, 0.25); padding-left: 64px; white-space: nowrap;">
                          child-1-1-1-1
                        </div>
                      </li>
                      <li class="cmpui_tree-view__item-dir">
                        <ol class="cmpui_tree-view__item-dir-ol">
                          <li class="cmpui_tree-view__item">
                            <div style="background-color: rgba(0, 0, 0, 0.3); padding-left: 80px; white-space: nowrap;">
                              child-1-1-1-1-1
                            </div>
                          </li>
                          <li class="cmpui_tree-view__item-dir">
                            <ol class="cmpui_tree-view__item-dir-ol">
                              <li class="cmpui_tree-view__item">
                                <div style="background-color: rgba(0, 0, 0, 0.35); padding-left: 96px; white-space: nowrap;">
                                  child-1-1-1-1-1-1
                                </div>
                              </li>
                              <li class="cmpui_tree-view__item-dir">
                                <ol class="cmpui_tree-view__item-dir-ol">
                                  <li class="cmpui_tree-view__item">
                                    <div style="background-color: rgba(0, 0, 0, 0.4); padding-left: 112px; white-space: nowrap;">
                                      child-1-1-1-1-1-1-1
                                    </div>
                                  </li>
                                  <li class="cmpui_tree-view__item-dir">
                                    <ol class="cmpui_tree-view__item-dir-ol">
                                      <li class="cmpui_tree-view__item">
                                        <div style="background-color: rgba(0, 0, 0, 0.45); padding-left: 128px; white-space: nowrap;">
                                          child-1-1-1-1-1-1-1-1
                                        </div>
                                      </li>
                                      <li class="cmpui_tree-view__item-dir">
                                        <ol class="cmpui_tree-view__item-dir-ol">
                                          <li class="cmpui_tree-view__item">
                                            <div style="background-color: rgba(0, 0, 0, 0.5); padding-left: 144px; white-space: nowrap;">
                                              child-1-1-1-1-1-1-1-1-1
                                            </div>
                                          </li>
                                          <li class="cmpui_tree-view__item-dir">
                                            <ol class="cmpui_tree-view__item-dir-ol">
                                              <li class="cmpui_tree-view__item">
                                                <div style="background-color: rgba(0, 0, 0, 0.55); padding-left: 160px; white-space: nowrap;">
                                                  child-1-1-1-1-1-1-1-1-1-1
                                                </div>
                                              </li>
                                            </ol>
                                          </li>
                                        </ol>
                                      </li>
                                    </ol>
                                  </li>
                                </ol>
                              </li>
                            </ol>
                          </li>
                        </ol>
                      </li>
                    </ol>
                  </li>
                </ol>
              </li>
            </ol>
          </li>
        </ol>
      </li>
    </ol>
  </div>
</div>
`;
